File: Demos/Scoreboard/app/src/main/java/com/codeingnight/android/scoreboard/MainActivity.java
package com.codeingnight.android.scoreboard;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.util.Log;

import com.codeingnight.android.scoreboard.databinding.ActivityMainBinding;

import java.util.Stack;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    ScoreboardViewModel viewModel;
    ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        viewModel = new ViewModelProvider(this).get(ScoreboardViewModel.class);
        binding.setViewModel(viewModel);
        binding.setLifecycleOwner(this);
    }
}

File: 01_AndroidUI/02_ViewBinding/app/src/main/java/com/codingnight/example/viewbinding/MainActivity.kt
package com.codingnight.example.viewbinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.codingnight.example.viewbinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.textView.text = "Text"
    }
}